A43-sdk-authorization.md

```diff
@@ -0,0 +1,707 @@
+		(resp interface{}, err error) {
+  // ...
+}
+
+// StreamInterceptor intercepts incoming Stream RPC requests.
+// Only authorized requests are allowed to pass. Otherwise, an unauthorized
+// error is returned to the client.
+func (i *StaticInterceptor) StreamInterceptor(
+	srv interface{}, ss grpc.ServerStream, info *grpc.StreamServerInfo, handler grpc.StreamHandler)
+		(err error) {
+  // ...
+}
+
+type FileWatcherInterceptor struct {
+  internalInterceptors StaticInterceptors
+  policyFile           string
+  policyContents       string
+  refreshDuration      time.Duration
+  cancel               context.CancelFunc
+}
+
+// NewFileWatcher returns a new FileWatcherInterceptor from a policy file
+// that contains JSON string of authorization policy and a refresh duration to
+// specify the amount of time between policy refreshes.
+func NewFileWatcher(file string, duration time.Duration) (*FileWatcherInterceptor, error);
+
+// run is a long running goroutine which watches for changes in file path, and
+// updates the internalInterceptors upon modification.
+func (i *FileWatcherInterceptor) run(ctx context.Context) {
+  // ...
+}
+
+// updateInternalInterceptors checks if the policy file that is watching has changed,
+// and if so, updates the internalInterceptors with the policy. Unlike the
+// constructor, if there is an error in reading the file or parsing the policy, the
+// previous internalInterceptors will not be replaced.
+func (i *FileWatcherInterceptor) updateInternalInterceptors() {
+  // ...
+}
+
+func (i *FileWatcherInterceptor) UnaryInterceptor(
+	ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) 
+		(resp interface{}, err error) {
+  // ...
+}
+
+func (i *FileWatcherInterceptor) StreamInterceptor(
+	srv interface{}, ss grpc.ServerStream, info *grpc.StreamServerInfo, handler grpc.StreamHandler) 
+		(err error) {
+  // ...
+}
+
+// Close cleans up resources allocated by the interceptors.
+func (i *FileWatcherInterceptor) Close() {
+  i.cancel()
+}
+```
+
+##### Example: Using a Static Authz Interceptor in gRPC-Go Server
+
+```Go
+creds := credentials.NewServerTLSFromFile(certFile, keyFile)
+i, err := authz.NewStatic(authzPolicy)
+serverOpts := []grpc.ServerOption{
+  grpc.Creds(creds),
+  grpc.ChainUnaryInterceptor(i.UnaryInterceptor),
+  grpc.ChainStreamInterceptor(i.StreamInterceptor),
+}
+s := grpc.NewServer(serverOpts...)
+```
+
+##### Example: Using a Dynamic File Watcher Authz Interceptor in gRPC-Go Server
+
+```Go
+creds := credentials.NewServerTLSFromFile(certFile, keyFile)
+i, err := authz.NewFileWatcher(authzPolicyFile, 1 * time.Hour)
+serverOpts := []grpc.ServerOption{
+  grpc.Creds(creds),
+  grpc.ChainUnaryInterceptor(i.UnaryInterceptor),
+  grpc.ChainStreamInterceptor(i.StreamInterceptor),
+}
+s := grpc.NewServer(serverOpts...)
+// In the end, free up the resources used for policy refresh.
+defer i.Close()
```

Comment:
This should be done immediately upon `NewFileWatcher`s return.  (After checking `err`)

Reply:
_After_ checking `err`.  You have this above now.